File: languages/php/Bloock/Entity/Identity/BooleanAttribute.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents an attribute with a boolean value.
 */
class BooleanAttribute extends Attribute
{
    public static function fromProto(\Bloock\BooleanAttribute $res): BooleanAttribute
    {
        return new BooleanAttribute($res->getId(), $res->getValue());
    }

    public function toProto(): \Bloock\BooleanAttribute
    {
        $p = new \Bloock\BooleanAttribute();
        $p->setId($this->id);
        $p->setValue($this->value);
        return $p;
    }
}

File: languages/python/bloock/entity/identity/boolean_attribute.py
from __future__ import annotations

import bloock._bridge.proto.bloock_identity_entities_pb2 as proto
from bloock.entity.identity.attribute import Attribute


class BooleanAttribute(Attribute):
    """
    Represents an attribute with a boolean value.
    """
    @staticmethod
    def from_proto(a: proto.BooleanAttribute) -> BooleanAttribute:
        return BooleanAttribute(
            id=a.id,
            value=a.value,
        )

    def to_proto(self) -> proto.BooleanAttribute:
        return proto.BooleanAttribute(
            id=self.id,
            value=self.value,
        )
